source/agora/common/Deserializer.d

```diff
@@ -168,48 +169,30 @@ public void deserializePart (T)(ref T record, scope DeserializeDg dg)
     @trusted
     if (is(T == enum))
 {
-    import std.traits;
     OriginalType!T orig_val;
     deserializePart(orig_val, dg);
     record = cast(T)(orig_val);
 }
 
 /// Ditto
-public void deserializePart (ref ubyte record, scope DeserializeDg dg)
+public void deserializePart (T) (ref T record, scope DeserializeDg dg,
+    CompactMode compact = CompactMode.Yes)
     @trusted
+    if (isScalarType!T)
 {
-    record = dg(record.sizeof)[0];
-}
-
-/// Ditto
-public void deserializePart (ref ushort record, scope DeserializeDg dg)
-    @trusted
-{
-    record = deserializeVarInt!ushort(dg);
-}
-
-/// Ditto
-public void deserializePart (ref uint record, scope DeserializeDg dg)
-    @trusted
-{
-    record = deserializeVarInt!uint(dg);
-}
-
-/// Ditto
-public void deserializePart (ref long record, scope DeserializeDg dg)
-    @trusted
-{
-    record = *cast(long*)(dg(record.sizeof).ptr);
-}
-
-/// Ditto
-public void deserializePart (ref ulong record, scope DeserializeDg dg, CompactMode compact = CompactMode.Yes)
-    @trusted
-{
-    if (compact == CompactMode.Yes)
-        record = deserializeVarInt!ulong(dg);
+    static if (is(Unqual!T == bool))
```

Comment:
Was `bool` not supported before? Might be good to add a test.

Reply:
Not it wasn't because it needs `!!` (`ubyte` does not convert to `bool` implicitly). Will add test.